pkg/controllers/nodeclass/status/ami_test.go

```diff
@@ -89,133 +91,426 @@ var _ = Describe("NodeClass AMI Status Controller", func() {
-			fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2%s/recommended/image_id", version, fmt.Sprintf("-%s", karpv1.ArchitectureArm64)): "ami-id-789",
-		}
+		})
+		It("Should resolve all AMIs with correct requirements for AL2023", func() {
+			awsEnv.SSMAPI.Parameters = map[string]string{
+				fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2023/x86_64/standard/recommended/image_id", k8sVersion): "ami-amd64-standard",
+				fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2023/x86_64/nvidia/recommended/image_id", k8sVersion):   "ami-amd64-nvidia",
+				fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2023/x86_64/neuron/recommended/image_id", k8sVersion):   "ami-amd64-neuron",
+				fmt.Sprintf("/aws/service/eks/optimized-ami/%s/amazon-linux-2023/arm64/standard/recommended/image_id", k8sVersion):  "ami-arm64-standard",
+			}
+			nodeClass.Spec.AMISelectorTerms = []v1.AMISelectorTerm{{Alias: "al2023@latest"}}
+			ExpectApplied(ctx, env.Client, nodeClass)
+			ExpectObjectReconciled(ctx, env.Client, statusController, nodeClass)
+			nodeClass = ExpectExists(ctx, env.Client, nodeClass)
 
-		awsEnv.EC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{
-			Images: []*ec2.Image{
+			Expect(len(nodeClass.Status.AMIs)).To(Equal(4))
+			Expect(nodeClass.Status.AMIs).To(ContainElements([]v1.AMI{
 				{
-					Name:         aws.String("test-ami-1"),
-					ImageId:      aws.String("ami-id-123"),
-					CreationDate: aws.String(time.Now().Format(time.RFC3339)),
-					Architecture: aws.String("x86_64"),
-					Tags: []*ec2.Tag{
-						{Key: aws.String("Name"), Value: aws.String("test-ami-1")},
-						{Key: aws.String("foo"), Value: aws.String("bar")},
+					Name: "amd64-standard",
+					ID:   "ami-amd64-standard",
+					Requirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelArchStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.ArchitectureAmd64},
+						},
+						{
+							Key:      v1.LabelInstanceGPUCount,
+							Operator: corev1.NodeSelectorOpDoesNotExist,
+						},
+						{
+							Key:      v1.LabelInstanceAcceleratorCount,
+							Operator: corev1.NodeSelectorOpDoesNotExist,
+						},
 					},
 				},
 				{
-					Name:         aws.String("test-ami-2"),
-					ImageId:      aws.String("ami-id-456"),
-					CreationDate: aws.String(time.Now().Add(time.Minute).Format(time.RFC3339)),
-					Architecture: aws.String("x86_64"),
-					Tags: []*ec2.Tag{
-						{Key: aws.String("Name"), Value: aws.String("test-ami-2")},
-						{Key: aws.String("foo"), Value: aws.String("bar")},
+					Name: "amd64-nvidia",
+					ID:   "ami-amd64-nvidia",
+					Requirements: []corev1.NodeSelectorRequirement{
+						{
+							Key:      corev1.LabelArchStable,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.ArchitectureAmd64},
+						},
+						{
+							Key:      v1.LabelInstanceGPUCount,
```

Comment:
do we need a Accelerator NotExists in Accelerator here?

Reply:
It wouldn't be incorrect, but I don't think it's necessary. AFAIK the two labels are mutually exclusive; the existence of one implies the absence of the other.